### ISSUE TITLE
Nullable reference types in ValueTuples produce CS8619 when using non-null value

### ISSUE THREAD

**Issue by GedasFX** [dotnet/roslyn#75526]
**Version Used**:

Tested on these two:
Compiler version: '4.10.0-3.24324.8 (04fb3916)'. Language version: 12.0.
Compiler version: '4.8.0-7.23558.1 (e0917286)'. Language version: preview.

**Steps to Reproduce**:

```
using System.Collections.Generic;
using System.Linq;


IReadOnlyList<SourceRootItem> items = [];

IEnumerable<(string? Value, IEnumerable<int> Results)> result1 = items.Select(q => (q.Number, q.SubItems.Select(_ => 0)));
IEnumerable<(string? Value, IEnumerable<int> Results)> result2 = items.Select(q => (default(string), q.SubItems.Select(_ => 0)));
IEnumerable<(string? Value, IEnumerable<int> Results)> result3 = [("Number", [123])];

class SourceRootItem
{
    public string Number { get; init; } = null!;
    public IReadOnlyList<object> SubItems { get; init; } = null!;
}
```
`result1` produces the warning. Seems to only be caused by the `Select` as `results3` does not produce warning

[Sharplab link](https://sharplab.io/#v2:CYLg1APgAgDABFAjAFgNwFgBQsGIHQAyAlgHYCOGmWWAkgEoCmAhsAPIkA2AnsQM4AuAHgDKAewCuAJwDGDOqNH8a/BgFsAfHCIrVvOAF44AbQC6lLEgDMggBRIYAfjgA1Jh3EMANLmul+mxl5xDn5eAEpNSQYgkMQDLR1ePGEGDgZpfhsyA00svAA5cVUAIwZJbzJk8WLlNSSUtIybAH0cuBgwzsorW3snV3cvH0E/AOjg0Ii4KJj+ACZ47Trk1PTM7P1c4AYAMyYJu0QOiqqaxJXGzNbN9s6w7sRrQ8cXNw9vHtG4QInwyPGQpZ4kYbAAiQolMqg7xGRBzSwmMJmajYBZiKSyeSKWqqLAAbywcCJCCB9jgENKkjgeLgAHMGPxUFoSNomQBfeIkYIcACElGJJLg9GYbE4PCIAkEomKACs1pphNUcXoafTGczWXAOYYuRxeZQ2VggA==)

**Diagnostic Id**:

```
warning CS8619: Nullability of reference types in value of type 'IEnumerable<(string Number, IEnumerable<int>)>' doesn't match target type 'IEnumerable<(string? Value, IEnumerable<int> Results)>'.
```

**Expected Behavior**:

`string` should not produce nullability error when target type is `string?`

**Actual Behavior**:

The behavior is inverted: I need to suppress non-null values with `!` even though they are not null to begin with.

**Comment by huoyaoyuan** [comment]
The behavior is a bit inconsistent here. There is nullable covariance for tuples and covariant interfaces, but not the combination of them. More direct explanation:

```csharp
string x1 = "a";
string? y1 = x1; // no warning
IEnumerable<string> x2 = ["a"];
IEnumerable<string?> y2 = x2; // no warning
(string, int) x3 = ("a", 1);
(string?, int) y3 = x3; // no warning
IEnumerable<(string, int)> x4 = [("a", 1)];
IEnumerable<(string?, int)> y4 = x4; // warning!
```

The cases in your examples are different. For result2, the type of `default(string)` expression is `string?`, and you can see the inferred type argument for `Select` is `(string? Value, IEnumerable<int> Results)` if you hover it in Visual Studio. For result3, it uses "target-typing", which means the type of RHS of assignment is decided by the declaration on LHS.

**Comment by huoyaoyuan** [comment]
This is a duplicate of #21752. Generic covariance isn't applied to value tuples because they are value types.

**Comment by jaredpar** [comment]
Closing as a duplicate.
